# Exercise 2

```json
Για τις Ασκήσεις που θα λύσετε θα χρησιμοποιήσετε τα ψηφία του Αριθμού Μητρώου Α.Μ.
Θεωρείστε το ΑΜ ως τετραψήφιο αριθμό με ψηφία: A B C D.
Δηλαδή ο ένας Φοιτητής με Α.Μ. 1343, θα έχει Α = 1, Β = 3, C = 4, D = 3

```

### Για το Α.Μ μου ισχύει: Α = 3, Β = 3, C = 2, D = 3

# Άσκηση 1

Κατά την διάρκεια του εργαστηρίου υλοποιήθηκε κύκλωμα για την μελέτη της διόδου και την δημιουργία της χαρακτηριστικής καμπύλης του στοιχείου. Συγκεκριμένα για την μελέτη της συμπεριφοράς της διόδου στην ορθή πόλωση χρησιμοποιήθηκε μια απλή δίοδος, ενώ για την μελέτη της ανάστροφης πόλωσης χρησιμοποιήθηκε δίοδος Zener. Καταγράψτε τις τιμές που λάβαμε από το πολύμετρο για την ένταση του ρεύματος ($I_D$) που διαρρέει την δίοδο και την τάση ($V_D$) στα άκρα της.

| A/A | VD (Ορθή πόλωση διόδου) | ΙD (Ορθή πόλωση διόδου) | VD (Ανάστροφη πόλωση διόδου) | ΙD (Ανάστροφη πόλωση διόδου) |
| --- | --- | --- | --- | --- |
| 1 | 0.2 | 0 | -0.98 | 0 |
| 2 | 0.3 | 0 | -1.98 | 0 |
| 3 | 0.4 | 0.011 | -2.95 | 0 |
| 4 | 0.45 | 0.042 | -3.73 | -0.023 |
| 5 | 0.48 | 0.089 | -4.20 | -0.075 |
| 6 | 0.5 | 0.141 | -4.45 | -0.148 |
| 7 | 0.52 | 0.221 | -4.60 | -0.231 |
| 8 | 0.55 | 0.413 | -4.68 | -0.319 |
| 9 | 0.58 | 0.766 | -4.75 | -0.41 |
| 10 | 0.6 | 1.143 | -4.79 | -0.503 |
| 11 | 0.63 | 2.021 |  |  |
| 12 |  |  |  |  |
| 13 |  |  |  |  |
| 14 |  |  |  |  |
|  |  |  |  |  |

---

A. Να σχεδιάσετε (είτε με κάποιο λογισμικό, είτε σε μιλιμετρέ χαρτί με το χέρι) την χαρακτηριστική καμπύλη της διόδου. Η καμπύλη της ορθής πόλωσης θα πρέπει να είναι στο 1ο τεταρτημόριο του καρτεσιανού συστήματος συντεταγμένων, ενώ η ανάστροφη πόλωση στο 3ο τεταρτημόριο.

Β. Μπορεί να καταλάβετε από την χαρακτηριστική καμπύλη αν η δίοδος που χρησιμοποιήσαμε είναι πυριτίου ή γερμανίου; Εξηγείστε.

Γ. Ποια πιστεύετε με βάση την γραφική παράσταση ότι είναι η τιμή της τάσης Zener ($V_z$).

In [3]:
import plotly.graph_objects as go

# Δημιουργία γραφήματος
fig = go.Figure()

# Δεδομένα από τον πίνακα
vd_forward = [0.2, 0.3, 0.4, 0.45, 0.48, 0.5, 0.52, 0.55, 0.58, 0.6, 0.63]
id_forward = [0, 0, 0.011, 0.042, 0.089, 0.141, 0.221, 0.413, 0.766, 1.143, 2.021]

vd_reverse = [-0.98, -1.98, -2.95, -3.73, -4.2, -4.45, -4.6, -4.68, -4.75, -4.79]
id_reverse = [0, 0, 0, -0.023, -0.075, -0.148, -0.231, -0.319, -0.41, -0.503]

# Ορθή πόλωση
fig.add_trace(go.Scatter(
    x=vd_forward, 
    y=id_forward, 
    mode='lines+markers', 
    name='Ορθή πόλωση',
    hoverinfo='text',
    hovertext=[f'Vd: {vd} V<br>Id: {id} A' for vd, id in zip(vd_forward, id_forward)]
))

# Ανάστροφη πόλωση
fig.add_trace(go.Scatter(
    x=vd_reverse, 
    y=id_reverse, 
    mode='lines+markers', 
    name='Ανάστροφη πόλωση',
    hoverinfo='text',
    hovertext=[f'Vd: {vd} V<br>Id: {id} A' for vd, id in zip(vd_reverse, id_reverse)]
))

# Προσθήκη τίτλων και ετικετών
fig.update_layout(
    title="Χαρακτηριστική καμπύλη διόδου",
    xaxis_title="Τάση Vd (V)",
    yaxis_title="Ρεύμα Id (A)",
    xaxis=dict(showline=True, zeroline=True),
    yaxis=dict(showline=True, zeroline=True),
    legend=dict(x=0.01, y=0.99)
)

# Εμφάνιση γραφήματος
fig.show()


---
## B:

Από τη χαρακτηριστική καμπύλη της διόδου, μπορούμε να προσδιορίσουμε αν η δίοδος είναι πυριτίου ή γερμανίου, βασιζόμενοι στην τάση κατωφλίου ($V_f$) στην ορθή πόλωση.

- **Δίοδος Πυριτίου**: Η τάση κατωφλίου βρίσκεται συνήθως γύρω στα $0.6- (0.63) \sim0.7 \, \text{V}$.
- **Δίοδος Γερμανίου**: Η τάση κατωφλίου είναι χαμηλότερη, περίπου $0.3 \, \text{V}$.

Στην καμπύλη της ορθής πόλωσης, παρατηρούμε ότι το ρεύμα αρχίζει να αυξάνεται σημαντικά από την περιοχή των $0.5-0.6 \, \text{V}$, ενώ η εκθετική αύξηση γίνεται εμφανής μετά τα $0.6 \, \text{V}$. Αυτό υποδεικνύει ότι η δίοδος που χρησιμοποιήθηκε είναι **πυριτίου**, καθώς η χαρακτηριστική της συμπεριφορά ταιριάζει με την αναμενόμενη για διόδους πυριτίου.

---


In [3]:
import plotly.graph_objects as go

# Δημιουργία γραφήματος
fig = go.Figure()

# Δεδομένα από τον πίνακα
vd_forward = [0.2, 0.3, 0.4, 0.45, 0.48, 0.5, 0.52, 0.55, 0.58, 0.6, 0.63]
id_forward = [0, 0, 0.011, 0.042, 0.089, 0.141, 0.221, 0.413, 0.766, 1.143, 2.021]

vd_reverse = [-0.98, -1.98, -2.95, -3.73, -4.2, -4.45, -4.6, -4.68, -4.75, -4.79]
id_reverse = [0, 0, 0, -0.023, -0.075, -0.148, -0.231, -0.319, -0.41, -0.503]

# # Ορθή πόλωση
# fig.add_trace(go.Scatter(
#     x=vd_forward, 
#     y=id_forward, 
#     mode='lines+markers', 
#     name='Ορθή πόλωση',
#     hoverinfo='text',
#     hovertext=[f'Vd: {vd} V<br>Id: {id} A' for vd, id in zip(vd_forward, id_forward)]
# ))

# Ανάστροφη πόλωση
fig.add_trace(go.Scatter(
    x=vd_reverse, 
    y=id_reverse, 
    mode='lines+markers', 
    name='Ανάστροφη πόλωση',
    hoverinfo='text',
    hovertext=[f'Vd: {vd} V<br>Id: {id} A' for vd, id in zip(vd_reverse, id_reverse)]
))

# Προσθήκη τίτλων και ετικετών
fig.update_layout(
    title="Χαρακτηριστική καμπύλη διόδου",
    xaxis_title="Τάση Vd (V)",
    yaxis_title="Ρεύμα Id (A)",
    xaxis=dict(showline=True, zeroline=True),
    yaxis=dict(showline=True, zeroline=True),
    legend=dict(x=0.01, y=0.99)
)

# Εμφάνιση γραφήματος
fig.show()




## Γ:



Η τάση Zener ($V_z$) είναι η τάση στην οποία η δίοδος Zener αρχίζει να αγώγιμα στην ανάστροφη πόλωση, δηλαδή το ρεύμα αυξάνεται σημαντικά σε απόλυτη τιμή. Από τη γραφική παράσταση παρατηρούμε ότι η αύξηση του ρεύματος γίνεται αισθητή περίπου στα $-4.2 \, \text{V}$.

Επομένως, μπορούμε να εκτιμήσουμε ότι η τιμή της τάσης Zener ($V_z$) είναι περίπου $-4.2 \, \text{V}$.

# **Άσκηση 2**

**Α) Να υπολογιστεί η κυμάτωση peak to peak (από κορυφή σε κορυφή) στα άκρα του φορτίου(**$R_L$) **. Το τροφοδοτικό χρησιμοποιεί κύκλωμα ημιανορθωτή ή ανορθωτή με βάση τα παρακάτω:**

**C** = 2

- Αν 0 ≤ **C** ≤ 3 τότε χρησιμοποιεί κύκλωμα ημιανορθωτή.

Δίνονται: ΝΠρωτ = (**A** + 1)*40, Nδευτ = (1 + **Α**)*10, Fin = (**D** + 1)*10 Hz, C1 = (**C** + 1)*10 μF,

R1 = R2 = RLOAD = (**B** + 2) KΩ, Vz = 5V.

![Διάγραμμα](./image%20(3).png)


Δίνονται:  $V_{\text{Rout}} = \frac{I_{\text{LOAD}}}{F_{\text{out}} \cdot C_1}, \quad \frac{V_{\text{πρωτ}}}{V_{\text{δευτερ}}} = \frac{N_{\text{πρωτ}}}{N_{\text{δευτερ}}}, \quad V_p = \frac{V_{\text{rms}}}{0.707}$

Β) Συνδέσαμε ένα κύκλωμα ημιανορθωτή σε κανάλι παλμογράφου και λάβαμε τις παρακάτω μετρήσεις. Στο πάνω αριστερά σχήμα φαίνεται το αρχικό σήμα εισόδου και στο πάνω δεξιά το σήμα εξόδου του ημιανορθωτή. Στη συνέχεια προσθέσαμε πυκνωτή (φίλτρο) εξομάλυνσης παράλληλα με την έξοδο του ημιανορθωτή και λάβαμε την κυματομορφή που φαίνεται στο σχήμα κάτω αριστερά. Λαμβάνοντας υπόψη τις κλίμακες του παλμογράφου υπολογίστε:

1. την συχνότητα του αρχικού σήματος
2. τον συντελεστή εξομάλυνσης *γ* του φίλτρου.

![Διάγραμμα](image%20(4).png)
![Διάγραμμα](image%20(5).png)

---

## A:

**Παράμετροι από τα δεδομένα:**

- $ΝΠρωτ = (3 + 1)*40 = 160 σπείρες$
- $Nδευτ = (1 + 3)*10 = 40 σπείρες$
- $Fin = (3 + 1)*10 = 40 Hz$
- $C1 = (2 + 1)*10 = 30 μF$
- $R1 = R2 = RLOAD = (3 + 2)*5000 = 5000 Ω$

**Ανάλυση τύπου ανορθωτή:**
Εφόσον C = 2 και 0 ≤ C ≤ 3, το κύκλωμα χρησιμοποιεί ημιανορθωτή.

**Υπολογισμός τάσεων:**

1. Λόγος μετασχηματιστή: $\frac{V_{\text{πρωτ}}}{V_{\text{δευτερ}}} = \frac{160}{40} = 4$
2. Για ημιανορθωτή:$Fout = Fin = 40 Hz$
- $ILOAD = Vz/RLOAD = 5V/5000Ω = 1mA$
1. Κυμάτωση: $V_{\text{Rout}} = \frac{I_{\text{LOAD}}}{F_{\text{out}} \cdot C_1} = \frac{0.001}{40 \cdot 30 \cdot 10^{-6}} = 0.833V$

Η κυμάτωση peak to peak στα άκρα του φορτίου είναι **0.833V**.

---

## Β_1

### Συχνότητα του αρχικού σήματος

**κλίμακα χρόνου: Δεν δίνετε αρά κλίμακα χρόνου = 1, ξέρω ότι το είχατε πει στο εργαστήριο αλλά δεν το θυμάμαι ούτε το έχω σημειωμένο κάπου** 

    • Χρόνος/τετράγωνο = 1ms/div

Η συχνότητα $f$ υπολογίζεται από την περίοδο $T$ με τον τύπο:
$f = \frac{1}{T}$
Από το διάγραμμα, η περίοδος του αρχικού σήματος είναι $T = 4 \, \text{ms} = 0.004 \, \text{s}$. Επομένως:
$f = \frac{1}{0.004} = 250 \, \text{Hz}$

## B_2

### Συντελεστής εξομάλυνσης $\gamma$

Ο συντελεστής εξομάλυνσης $\gamma$ δίνεται από τη σχέση:
$\gamma = \frac{\Delta V_{\text{smooth}}}{\Delta V}$
όπου:

- $\Delta V_{\text{smooth}} = 0.2 \, \text{V}$: το πλάτος της κυμάτωσης του εξομαλυνμένου σήματος (ripple).
- $\Delta V = 2.0 \, \text{V}$: το πλάτος του ανορθωμένου σήματος χωρίς εξομάλυνση.

Υπολογίζουμε:
$\gamma = \frac{0.2}{2.0} = 0.1$

---

→Συχνότητα του αρχικού σήματος: **250 Hz**.

→Συντελεστής εξομάλυνσης: **0.1**.
---

# **Άσκηση 3**

Κατά τη διάρκεια ενός πειράματος στο εργαστήριο υλοποιήθηκε το κύκλωμα (σχήμα 1) για την εξαγωγή της χαρακτηριστικής καμπύλης εξόδου του διπολικού τρανζίστορ επαφής.

![Διάγραμμα](image%20(6).png)

Σχήμα 1: Κύκλωμα πόλωσης τρανζίστορ κοινού εκπομπού με πόλωση βάσης, για την εξαγωγή της χαρακτηριστικής καμπύλης εξόδου

Στο παραπάνω κύκλωμα πόλωσης βάσης, τα μεγέθη για τις αντιστάσεις και τις τάσεις που χρησιμοποιήθηκαν κατά τη διάρκεια του εργαστηρίου είναι R1 = 1ΜΩ, R2 = 1ΚΩ και VBB = 5V. Με δεδομένο ότι Vcc = 10V και ο συντελεστής ενίσχυσης (κέρδος) του τρανζίστορ είναι β = 500, να γίνει η DC ανάλυση του κυκλώματος:

Α) **Υπολογίζοντας θεωρητικά το σημείο λειτουργίας Q**

Β) **Σχεδιάζοντας τη γραμμή φορτίου** επάνω στη χαρακτηριστική καμπύλη. Σε ποια περιοχή λειτουργίας βρίσκεται το τρανζίστορ (δώστε απάντηση με βάση τη θέση του σημείου Q στη χαρακτηριστική καμπύλη εξόδου);

In [6]:
# Given values
R1 = 1e6  # 1 MΩ
R2 = 1e3  # 1 kΩ
VBB = 5  # 5 V
Vcc = 10  # 10 V
beta = 500  # Current gain of the transistor

# Base current calculation
IB = (VBB - 0.7) / R1  # Subtracting 0.7V for the base-emitter voltage drop

# Collector current calculation
IC = beta * IB

# Voltage across R2
VR2 = IC * R2

# Collector-emitter voltage
VCE = Vcc - VR2

print(f"IB is:{IB}\nIC is: {IC}\nVCE is: {VCE}")

IB is:4.2999999999999995e-06
IC is: 0.0021499999999999996
VCE is: 7.8500000000000005


In [2]:
import numpy as np
import plotly.graph_objects as go

# -----------------------------------
# Παράμετροι τρανζίστορ και κυκλώματος
# -----------------------------------
beta = 500        # Συντελεστής ενίσχυσης
R2 = 1e3          # 1 kΩ
VCC = 10.0        # 10 V
VCE_sat = 0.2     # Τάση κορεσμού (προσεγγιστικά)
# Ρεύματα βάσης (διάφορες τιμές για τις καμπύλες)
Ib_values = [5e-6, 10e-6, 15e-6, 20e-6, 25e-6]

# -----------------------------------
# Προετοιμασία "άξονα" VCE
# -----------------------------------
VCE_points = np.linspace(0, 10, 200)  # από 0 έως 10 V

# Δημιουργία γραφήματος
fig = go.Figure()

# -----------------------------------
# Σχεδίαση χαρακτηριστικών εξόδου
# -----------------------------------
for Ib in Ib_values:
    Ic_curve = []
    for vce in VCE_points:
        if vce <= VCE_sat:
            Ic_curve.append(beta * Ib)
        else:
            Ic_curve.append(beta * Ib)
    fig.add_trace(go.Scatter(
        x=VCE_points, 
        y=Ic_curve, 
        mode='lines', 
        name=f'Ib = {Ib*1e6:.0f} μA',
        hoverinfo='text',
        hovertext=[f'VCE: {vce:.2f} V<br>IC: {ic:.6f} A' for vce, ic in zip(VCE_points, Ic_curve)]
    ))

# -----------------------------------
# Σχεδίαση ευθείας φορτίου (Load Line)
# -----------------------------------
Ic_load_line = (VCC - VCE_points) / R2
fig.add_trace(go.Scatter(
    x=VCE_points, 
    y=Ic_load_line, 
    mode='lines', 
    name='Load Line',
    line=dict(dash='dash'),
    hoverinfo='text',
    hovertext=[f'VCE: {vce:.2f} V<br>IC: {ic:.6f} A' for vce, ic in zip(VCE_points, Ic_load_line)]
))

# -----------------------------------
# Υπολογισμός σημείου λειτουργίας (Q-point)
# -----------------------------------
R1 = 1e6
VBB = 5.0
VBE = 0.7
Ib = (VBB - VBE) / R1  # ρεύμα βάσης
IcQ = beta * Ib        # ρεύμα συλλέκτη
VCEQ = VCC - IcQ * R2  # τάση συλλέκτη-εκπομπού

# Σχεδιάζουμε το σημείο Q
fig.add_trace(go.Scatter(
    x=[VCEQ], 
    y=[IcQ], 
    mode='markers+text', 
    name='Q-point',
    text=[f'Q\n(VCE={VCEQ:.2f} V,\nIC={IcQ*1e3:.2f} mA)'],
    textposition='top right',
    marker=dict(color='black', size=10)
))

# Προσθήκη τίτλων και ετικετών
fig.update_layout(
    title="Χαρακτηριστικές εξόδου τρανζίστορ με την ευθεία φορτίου",
    xaxis_title="VCE (V)",
    yaxis_title="IC (A)",
    xaxis=dict(showline=True, zeroline=True),
    yaxis=dict(showline=True, zeroline=True),
    legend=dict(x=1.05, y=1)  # Move the legend outside the main graph area
)

# Εμφάνιση γραφήματος
fig.show()



---
# **Δεδομένα Κυκλώματος**
Ακολουθούν τα δεδομένα που δόθηκαν στην εκφώνηση και χρησιμοποιήθηκαν για την επίλυση:

| Παράμετρος      | Τιμή                 |
|------------------|----------------------|
| $$ R_1 $$       | $$ 1 \, M\Omega $$  |
| $$ R_2 $$       | $$ 1 \, k\Omega $$  |
| $$ V_{BB} $$    | $$ 5 \, V $$         |
| $$ V_{CC} $$    | $$ 10 \, V $$        |
| $$ V_{BE} $$    | $$ 0.7 \, V $$       |
| Συντελεστής β   | $$ 500 $$            |

---

#### **Υπολογισμοί Βήμα προς Βήμα**

1. **Υπολογισμός Ρεύματος Βάσης ($ I_B $):**

   Το ρεύμα βάσης υπολογίζεται από τη σχέση:
   $
   I_B = \frac{V_{BB} - V_{BE}}{R_1}
   $

   Αντικαθιστώντας τις τιμές:
   $
   I_B = \frac{5 - 0.7}{1 \cdot 10^6} = 4.3 \, \mu A
   $

2. **Υπολογισμός Ρεύματος Συλλέκτη ($ I_C $):**

   Το ρεύμα συλλέκτη σχετίζεται με το ρεύμα βάσης μέσω του συντελεστή ενίσχυσης ($ β $):
   $
   I_C = β \cdot I_B
   $

   Αντικαθιστώντας τις τιμές:
   $
   I_C = 500 \cdot (4.3 \cdot 10^{-6}) = 2.15 \, mA
   $

3. **Υπολογισμός Τάσης Συλλέκτη-Εκπομπού ($ V_{CE} $):**

   Η τάση συλλέκτη-εκπομπού υπολογίζεται από τη σχέση:
   $
   V_{CE} = V_{CC} - I_C R_2
   $

   Αντικαθιστώντας τις τιμές:
   $
   V_{CE} = 10 - (2.15 \cdot 10^{-3}) \cdot (1 \cdot 10^3) = 7.85 \, V
   $

4. **Γραμμή Φορτίου:**

   Η γραμμή φορτίου περιγράφεται από την εξίσωση:
   $
   I_C = \frac{V_{CC} - V_{CE}}{R_2}
   $

   Τα δύο χαρακτηριστικά σημεία της γραμμής φορτίου είναι:
   
   - Όταν $ V_{CE} = 0\,V$: 
     $
     I_C = I_{C,\text{max}} = \frac{V_{CC}}{R_2} = \frac{10}{1\cdot10^3} = 10\,mA
     $
     
   - Όταν $ I_C = 0\,A$: 
     $
     V_{CE} = V_{CC} = 10\,V
     $

---

#### **Σημείο Λειτουργίας Q**
Το σημείο λειτουργίας Q βρίσκεται στη διασταύρωση της γραμμής φορτίου με την χαρακτηριστική καμπύλη εξόδου του τρανζίστορ.

- Το σημείο Q είναι: 
$
Q(V_{CE}, I_C) = (7.85\,V, 2.15\,mA)
$

---

#### **Περιοχή Λειτουργίας**
Το τρανζίστορ λειτουργεί στην *ενεργή περιοχή* (active region), διότι:

- Η τάση $ V_{CE} > V_{BE} > 0.7\,V$.
- Υπάρχει σημαντικό $ I_C > 0\,A$, αλλά δεν έχει φτάσει στον κόρο.

Η θέση του σημείου Q πάνω στη χαρακτηριστική καμπύλη επιβεβαιώνει ότι το τρανζίστορ βρίσκεται στην ενεργή περιοχή.

---

#### **Διάγραμμα Γραμμής Φορτίου**
Στο συνημμένο διάγραμμα:

- Η μπλε γραμμή είναι η γραμμή φορτίου.
- Το κόκκινο σημείο αντιπροσωπεύει το σημείο λειτουργίας Q: $ (7.85\,V, 2.15\,mA) $.

